### PR TITLE
fix: add query to get related tags for source

### PR DESCRIPTION
**Pull request by rebelchris**
Add query to sources to get relatedTags for the source.
Based on:
- last 90 days post
- keyword count
- allowed keywords

Screenshot 2024-04-11 at 16 16 23
(subnote: would actually be nice to have a postman collection shared 😅)

### CODE REVIEW

File: src/schema/sources.ts [hunk 1]
```diff
@@ -1121,6 +1138,30 @@ export const resolvers: IResolvers<any, Context> = {
       }
       return res[0];
     },
+    relatedTags: async (_, { sourceId }, ctx): Promise<GQLTagResults> => {
```
Comment (rebelchris) [inline src/schema/sources.ts:1141]: @idoshamun Mostly from query point of view, any suggestions here or this will do for now?

Comment (idoshamun) [reply src/schema/sources.ts:1141]: that looks ok! let's monitor it and if needed introduce redis in front

**Review by idoshamun — APPROVED** [review]

File: src/schema/sources.ts (hunk 1 above)
Comment (rebelchris) [reply src/schema/sources.ts:1141]: We can also consider some graphQl magic on FE with high stale time and combing the queries into one request.

Comment (capJavert) [reply src/schema/sources.ts:1141]: For tag recommendation of onboarding we used materialized view which we refresh in cron once a week.

File: src/schema/sources.ts [hunk 2]
```diff
@@ -1121,6 +1138,30 @@ export const resolvers: IResolvers<any, Context> = {
...
+        .createQueryBuilder()
+        .from(Post, 'p')
+        .select('pk.keyword, count(pk.keyword) as count')
+        .innerJoin(
+          PostKeyword,
+          'pk',
+          'pk.postId = p.id and pk.status = :status',
+          { status: 'allow' },
+        )
+        .where({ sourceId })
```
Comment (capJavert) [inline src/schema/sources.ts:1153]: Do we need to check if source access is allowed?

Comment (rebelchris) [reply src/schema/sources.ts:1153]: Nah think its fine, doesnt really expose anything and would cost another query 🥲

File: src/schema/sources.ts (hunk 1 above)
Comment (rebelchris) [reply src/schema/sources.ts:1141]: We can start with this and monitor if we need more if that works for you?

Comment (capJavert) [reply src/schema/sources.ts:1141]: Yeah, from my side this looks ok for now.

File: src/schema/sources.ts (hunk 2 above)
Comment (capJavert) [reply src/schema/sources.ts:1153]: Ok, let's leave it as is for now.

**Review by capJavert — APPROVED** [review]

**Pull request merged**